.travis.yml

```diff
@@ -33,34 +40,74 @@ jobs:
     - stage: test with coverage
       php: 7.1
       before_install:
+        - stty cols 130
         - echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
         - echo 'opcache.enable_cli=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
         - travis_retry composer self-update
-      script: vendor/bin/phpunit -c phpunit.xml.dist --colors --verbose --coverage-text --coverage-clover=coverage.clover
+      script: vendor/bin/phpunit -c phpunit.xml.dist --colors --columns 121 --verbose --coverage-text --coverage-clover=coverage.clover
       after_success:
         - wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.clover
         - bash <(curl -s https://codecov.io/bash) -f coverage.clover
+    - stage: composer validate
+      php: 7.1
+      before_install:
+        - stty cols 130
+        - phpenv config-rm xdebug.ini || echo "xdebug not available"
+        - echo 'opcache.enable=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+        - echo 'opcache.enable_cli=1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
+      install: skip
+      script: composer validate --strict
     - stage: check coding style
       php: 7.1
       script: php -n -d memory_limit=768M vendor/bin/php-cs-fixer fix --dry-run -vv
     - stage: static code analysis
       php: 7.1
       script: vendor/bin/phpstan analyse -l 7 -c phpstan.neon --autoload-file=vendor/autoload.php --memory-limit=768M --no-progress src tests/UserAgentsTest tests/BrowscapTest tests/fixtures
-    - stage: integration test
+    - stage: integration tests
+      php: 7.1
+      env:
+        - parser="browscap-php 4.x@dev"
```

Comment:
Should these be added to an allow_failure block in travis?

Or are we expecting that a fail in the 4.x parser should break a build?

Reply:
I would like it, that we handle version 4 the same as the current version.